### ISSUE TITLE
[BUG] KeyDB 6.2.1 does not compile on macOS

### ISSUE THREAD

**Issue by alebcay** [EQ-Alpha/KeyDB#384]
**Describe the bug**
Building KeyDB 6.2.1 results in the following error on macOS:
```
clang -o luac  luac.o print.o liblua.a -lm 
      CC ae.o
      CC adlist.o
      CC quicklist.o
      CC anet.o
      CC server.o
      CC dict.o
      CC sds.o
      CC zmalloc.o
      CC lzf_c.o
      CC lzf_d.o
      CC pqsort.o
      CC zipmap.o
      CC sha1.o
      CC ziplist.o
      CC release.o
      CC networking.o
      CC util.o
      CC object.o
      CC db.o
      CC replication.o
      CC rdb.o
  server.cpp:6565:9: error: use of undeclared identifier 'clock_nanosleep'
          clock_nanosleep(CLOCK_MONOTONIC, 0, &delay, NULL);
          ^
  1 error generated.
  make[1]: *** [server.o] Error 1
  make[1]: *** Waiting for unfinished jobs....
  make: *** [install] Error 2
```

**To reproduce**

Download and unpack the source for KeyDB 6.2.1, then `make install PREFIX=<an arbitrary dir>`.

**Expected behavior**

The build should succeed and KeyDB 6.2.1 should be installed to the specified prefix.

**Additional information**

This issue seems to stem from https://github.com/EQ-Alpha/KeyDB/commit/b03eab2a0628df157c1f6b6242bb500ca826ef45, when a usage of `clock_nanosleep()` was introduced in `src/server.cpp`, which is not available on macOS. It looks like macOS does have a [`nanosleep()`](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/nanosleep.2.html) function and a `mach_wait_until()` function available but it's unclear to me if either are a drop-in replacement.

**Comment by VivekSainiEQ** [comment]
Hi @alebcay,

Good catch! The problem has been fixed in commit 7a32ec39fdb738e9c3cd2b73ee18355ced793a65, by using nanosleep if we're on MacOS, and should be included in the next release.

**Comment by alebcay** [comment]
Thanks, confirmed that the problem is fixed when building 6.2.1 with the commit you linked applied on top.

**Comment by benschermel** [comment]
Closing this issue. Confirmed resolved in 6.2.2 and unstable branch.
